commands/createfixpullrequests.go

```diff
@@ -350,123 +303,13 @@ func (cfp *CreateFixPullRequestsCmd) updatePackageToFixedVersion(packageType cor
-	updateVersionCmd := fmt.Sprintf("mvn %s", strings.Join(updateVersionArgs, " "))
-	log.Debug(fmt.Sprintf("Running '%s'", updateVersionCmd))
-	updateVersionOutput, err := exec.Command("mvn", updateVersionArgs...).CombinedOutput() // #nosec G204
-	if err != nil {
-		return fmt.Errorf("mvn command failed: %s\n%s", err.Error(), updateVersionOutput)
-	}
-
-	// Update properties that represent this package's version.
-	for _, property := range properties {
-		updatePropertyArgs := []string{"-B", "versions:set-property", "-Dproperty=" + property, "-DnewVersion=" + fixVersion, "-DgenerateBackupPoms=false"}
-		updatePropertyCmd := fmt.Sprintf("mvn %s", strings.Join(updatePropertyArgs, " "))
-		log.Debug(fmt.Sprintf("Running '%s'", updatePropertyCmd))
-		updatePropertyOutput, err := exec.Command("mvn", updatePropertyArgs...).CombinedOutput() // #nosec G204
-		if err != nil {
-			return fmt.Errorf("mvn command failed: %s\n%s", err.Error(), updatePropertyOutput)
-		}
-	}
-
-	return nil
-}
-
-func fixPackageVersionPip(impactedPackage, fixVersion, requirementsFile string) error {
-	// This function assumes that the version of the dependencies is statically pinned in the requirements file or inside the 'install_requires' array in the setup.py file
-	fixedPackage := impactedPackage + "==" + fixVersion
-	if requirementsFile == "" {
-		requirementsFile = "setup.py"
-	}
-	wd, err := os.Getwd()
-	if err != nil {
-		return err
-	}
-	fullPath := filepath.Join(wd, requirementsFile)
-	if !strings.HasPrefix(filepath.Clean(fullPath), wd) {
-		return errors.New("wrong requirements file input")
-	}
-	data, err := os.ReadFile(filepath.Clean(requirementsFile))
-	if err != nil {
-		return err
-	}
-	currentFile := string(data)
-	// This regex will match the impactedPackage with it's pinned version e.g. PyJWT==1.7.1
-	re := regexp.MustCompile(pythonPackageRegexPrefix + impactedPackage + pythonPackageRegexSuffix)
-	packageToReplace := re.FindString(currentFile)
-	if packageToReplace == "" {
-		return fmt.Errorf("impacted package %s not found, fix failed", packageToReplace)
-	}
-	fixedFile := strings.Replace(currentFile, packageToReplace, fixedPackage, 1)
-	err = os.WriteFile(requirementsFile, []byte(fixedFile), 0600)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	packageHandler := packagehandlers.GetCompatiblePackageHandler(fixVersionInfo, cfp.details, &cfp.mavenDepToPropertyMap)
+	return packageHandler.UpdateImpactedPackage(impactedPackage, fixVersionInfo)
 }
 
-func fixPackageVersionPoetry(impactedPackage, fixVersion string) error {
-	// Install the desired fixed version
-	err := fixPackageVersionGeneric(coreutils.Poetry.GetExecCommandName(), []string{"add"}, impactedPackage, fixVersion, "==")
-	if err != nil {
-		return err
-	}
-	// Update Poetry lock file as well
-	return runPackageMangerCommand(coreutils.Poetry.GetExecCommandName(), []string{"update"})
+// TODO check this to be more generic!
```

Comment:
@eyalbe4 here I wasn't sure that its generic enough.
Maybe we could just check that:
if impact graph's length is exactly 2  == direct 
else == indirect

Reply:
I think that the below code is good.